### PR TITLE
Single admin page for DegreeMarketingData(Program) model

### PR DESCRIPTION
**Pull request by yro**
### Description
Implement single inheritance model for `DegreeMarketingData` table in model -- add single marketing admin page / model (and inherited model) for `DegreeMarketingData` that's wholly separate from the `Degree` model.

```
Program -- DegreeMarketing
    |           |
Degree--------- |
```

This introduces a sort of circular 1:1 dependency (as `DegreeMarketingData` is, in fact, an augmented `Program` model), we wished to protect against creating DegreeMarketingData without an associated Degree model.

This drops the prior `DegreeMarketing` model/table and adds an inheritance model `DegreeMarketingData`, as well as migration adding `Masters` to the `ProgramType` model (as `XSeries` and `MicroMasters` are added)

### Testing
- [ ] Unit tests
- [ ] Test Migrations

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @iloveagent57 @schenedx 

FYI: @edx/educator-neem

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

**Comment by schenedx** [comment]
FYI @mikedikan and @bderusha You guys should be aware of this change. Let us know if you don't agree with this approach.

**Comment by schenedx** [comment]
@yro Can you post a screenshot of what the new django admin page looks like?

**Comment by yro** [comment]
discovery-admin

**Comment by yro** [comment]
discovery-admin

**Comment by bderusha** [comment]
@schenedx I should have time this afternoon

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -1279,15 +1279,35 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
...
+class Curriculum(TimeStampedModel):
+    """
+    Curriculum 'master' model linking degree requirements and templates
     """
     uuid = models.UUIDField(blank=True, default=uuid4, editable=False, unique=True, verbose_name=_('UUID'))
     name = models.CharField(max_length=255, unique=True)
-    program = models.OneToOneField(Program, on_delete=models.CASCADE, related_name='degree')
+    degree = models.OneToOneField(Degree, on_delete=models.CASCADE, related_name='curriculum')
     program_curriculum = models.ManyToManyField(
-        Program, through='course_metadata.DegreeProgramCurriculum', related_name='degree_program_curricula'
```
Comment (iloveagent57) [inline course_discovery/apps/course_metadata/models.py]: This really should be a Many-to-many link out to `Program`, not degree.

File: course_discovery/apps/course_metadata/admin.py [hunk 2]
```diff
@@ -294,30 +294,42 @@ class DegreeCourseCurriculumInline(admin.TabularInline):
...
+    readonly_fields = ('uuid', )
+    search_fields = ('title', 'partner', 'marketing_slug')
+
+    # ordering the field display on admin page.
+    fields = (
+        'type', 'uuid', 'title', 'subtitle', 'status', 'partner', 'banner_image', 'banner_image_url', 'card_image_url',
+        'marketing_slug', 'overview', 'total_hours_of_effort', 'weeks_to_complete', 'min_hours_effort_per_week',
+        'max_hours_effort_per_week', 'authoring_organizations', 'hidden', 'faq', 'individual_endorsements',
+        'job_outlook_items', 'expected_learning_items', 'instructor_ordering',
+        # The fields below are explicitly on the 'DegreeMarketingData' model
```
Comment (iloveagent57) [inline course_discovery/apps/course_metadata/admin.py:330]: Don't worry about changing this, I can grab it in my next PR: there is no `DegreeMarektingData` model anymore.

File: course_discovery/apps/course_metadata/models.py [hunk 3]
```diff
@@ -1279,13 +1279,32 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
             super(Program, self).save(*args, **kwargs)
 
 
-class Degree(TimeStampedModel):
+class Degree(Program):
     """
-    Degree 'master' model linking requirements and templates
+    Marketing model for degree landing pages
+    Program inheritance model
```
Comment (iloveagent57) [inline course_discovery/apps/course_metadata/models.py:1285]: Nit for next PR: this line is already stated in the definition of the class.

File: course_discovery/apps/course_metadata/models.py [hunk 4]
```diff
@@ -1279,13 +1279,32 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
...
+    """
+    application_deadline = models.CharField(
+        help_text=_('String-based deadline field (e.g. FALL 2020)'),
+        max_length=255,
+    )
+    apply_url = models.CharField(
+        help_text=_('Callback URL to partner application flow'), max_length=255, blank=True)
+
+    class Meta(object):
+        verbose_name_plural = "degree marketing data"
```
Comment (iloveagent57) [inline course_discovery/apps/course_metadata/models.py:1295]: I also don't love this as a plural form of "Degree".

File: course_discovery/apps/course_metadata/models.py [hunk 5]
```diff
@@ -1279,13 +1279,32 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
...
+
+    def __str__(self):
+        return str('Marketing-specific fields for {}'.format(self.title))
+
+
+class Curriculum(TimeStampedModel):
+    """
+    Curriculum 'master' model linking degree requirements and templates
```
Comment (iloveagent57) [inline course_discovery/apps/course_metadata/models.py:1303]: This string could be better (e.g. what's a "template"?)

**Review by iloveagent57 — APPROVED** [review]
Looks good.  I have complaints about comments, but don't make changes here.  After this merges, I'll address them in my PR.

**Pull request merged**